Add Amazon Linux 2 build support

pgroudas opened the pull request:
### Add top level .gitignore

This just ignores intelliJ config files

### Add Amazon Linux 2 Build

- "Install_amazon-linux-2.sh" will install required build tools and
  development libraries.

- Adds "Makefile_amazon-linux-2" to describe the required dependency
  compilation order.

- Updates individual recipes for compiling each dependency to generally
  be more consistent.

- Fixes all dependencies to a specific version.

### Remove unused scripts.

We only support Amazon Linux 2

### Document build instructions.

### Update README

### Reset to specific tags / revisions

This just ensures that the way we manipulate the dependencies' git
repositories is consistent.

### make target to create archive of binary artifacts

### Set boost libdir for mcrouter build

The mcrouter configure script does not look for boost via
LD_LIBRARY_PATH nor CPPFLAGS '-H' value, instead it has a dedicated
cmdline option.

### Fail on systems with insufficient memory.

[Clubhouse Story pgroudas/amazon-linux-support](https://app.clubhouse.io/internal/story/pgroudas/amazon-linux-support)

Risk is Low

Commit:
Add top level .gitignore

This just ignores intelliJ config files

Commit:
Add Amazon Linux 2 Build

- "Install_amazon-linux-2.sh" will install required build tools and
  development libraries.

- Adds "Makefile_amazon-linux-2" to describe the required dependency
  compilation order.

- Updates individual recipes for compiling each dependency to generally
  be more consistent.

- Fixes all dependencies to a specific version.

Commit:
Remove unused scripts.

We only support Amazon Linux 2

Commit:
Document build instructions.

Commit:
Update README

favila commented on mcrouter/scripts/recipes/glog.sh:
```diff
@@ -8,12 +8,16 @@ source common.sh
+
+# Use a known compatible version
+# N.B. More recent versions (v0.5.0.rcx) failed to work with mcrouter.
+git reset --hard v0.4.0
```
Here you use a `reset`, elsewhere you use a `checkout`. I don't know if there's reason to prefer one over the other or if the inconsistency even matters--just pointing it out.

pgroudas replied:
Thank's for pointing this out.  I've gone ahead and made these all consistent.

favila reviewed (approved):
I know a lot of sweat went into this, good job!

Commit:
Reset to specific tags / revisions

This just ensures that the way we manipulate the dependencies' git
repositories is consistent.

Commit:
make target to create archive of binary artifacts

Commit:
Set boost libdir for mcrouter build

The mcrouter configure script does not look for boost via
LD_LIBRARY_PATH nor CPPFLAGS '-H' value, instead it has a dedicated
cmdline option.

Commit:
Fail on systems with insufficient memory.

Commit:
Fix typo in comment.
